### ISSUE TITLE
new version and return a 502 status code

### ISSUE THREAD

**Issue by hyd998877** [Mikubill/sd-webui-controlnet#626]
I encountered an error where the frontend run/predict/ endpoint would return a 502 status code. After inspecting my code, I found that the following line was causing the issue:
`btn.click(fn=my_class.save_submit_img2img_tab, inputs=gr.State(component), outputs=[])`
I was able to resolve the issue by changing the inputs argument to a list, like this:
`btn.click(fn=my_class.save_submit_img2img_tab, inputs=[component], outputs=[])`
I'm wondering why this change fixed the issue - is it a bug, or is there a reason why gr.State() shouldn't be used in this case?

**Comment by ljleb** [comment]
I cannot find this line in the code... Did the update break seemingly unrelated code (on your side) that was working before?

Usually you pass `gr.State` directly to a gradio event subscriber to pass a value directly to the function, instead of passing a component from which will be extracted the value passed to the function.

**Comment by hyd998877** [comment]
I have located the issue. It appears that the problem was caused by using a proxy server to access the Gradio web UI. Specifically, the app was deployed on a local network, and I was attempting to access it through a proxy on that network. The resulting 502 error was actually a gateway timeout error, which indicates that there may be an issue with how Gradio handles requests that come through a proxy.
